### PR TITLE
ARROW-2977: [Packaging] Release verification script should check rust too

### PR DESCRIPTION
**Pull request by kszucs**
I've found a couple of issues with the verification scripts:
1. The standalone js verification script seems obsolete
2. The windows script only checks arrow-cpp, parquet-cpp and pyarrow
3. The windows script doesn't create conda env

For the next release it'd nice to have consistent scripts on each platform (c_glib requires additional configuration on OSX).

### CODE REVIEW

**Comment by wesm** [comment]
JavaScript has been releasing separately

**Comment by kszucs** [comment]
Seems identical with the [test_js](https://github.com/apache/arrow/blob/master/dev/release/verify-release-candidate.sh#L210)

**Comment by wesm** [comment]
It would be a good idea to refactor the scripts to share code. The JavaScript release verification script is intended to verify the JavaScript source tarballs which are structurally different than the main releases

**Comment by wesm** [comment]
I opened https://issues.apache.org/jira/browse/ARROW-2984

**Review by wesm — APPROVED** [review]
+1

**Comment by kszucs** [comment]
Ok, thanks!

File: dev/release/verify-release-candidate.sh [hunk 1]
```diff
@@ -225,6 +225,29 @@ test_js() {
...
+  export RUSTUP_HOME=`pwd`/test-rustup
+  export CARGO_HOME=`pwd`/test-rustup
+
+  curl https://sh.rustup.rs -sSf | sh -s -- -y
+  source $RUSTUP_HOME/env
+
+  # build and test rust
+  pushd rust
+
+  # raises on any formatting errors (disabled, because RC1 has a couple)
```
Comment (kszucs) [inline dev/release/verify-release-candidate.sh]: Here is the [fix](https://github.com/apache/arrow/commit/889e1e6cde9455d1ed1dd6b827909fbeec0b1726)

**Pull request closed without merging**